header/header_modifier.go

```diff
@@ -27,31 +27,52 @@ func init() {
 	parse.Register("header.Modifier", modifierFromJSON)
 }
 
-type modifier struct {
+type HeaderSetBehavior int
+
+const (
+	ReplaceValues HeaderSetBehavior = iota
+	AppendValues
+)
+
+type Modifier struct {
 	name, value string
+	behavior    HeaderSetBehavior
 }
 
 type modifierJSON struct {
-	Name  string               `json:"name"`
-	Value string               `json:"value"`
-	Scope []parse.ModifierType `json:"scope"`
+	Name     string               `json:"name"`
+	Value    string               `json:"value"`
+	Behavior string               `json:"behavior"`
+	Scope    []parse.ModifierType `json:"scope"`
 }
 
 // ModifyRequest sets the header at name with value on the request.
-func (m *modifier) ModifyRequest(req *http.Request) error {
-	return proxyutil.RequestHeader(req).Set(m.name, m.value)
+func (m *Modifier) ModifyRequest(req *http.Request) error {
+	h := proxyutil.RequestHeader(req)
+	if m.behavior == AppendValues {
+		return h.Add(m.name, m.value)
+	}
+	return h.Set(m.name, m.value)
 }
 
 // ModifyResponse sets the header at name with value on the response.
-func (m *modifier) ModifyResponse(res *http.Response) error {
-	return proxyutil.ResponseHeader(res).Set(m.name, m.value)
+func (m *Modifier) ModifyResponse(res *http.Response) error {
+	h := proxyutil.ResponseHeader(res)
+	if m.behavior == AppendValues {
+		return h.Add(m.name, m.value)
+	}
+	return h.Set(m.name, m.value)
+}
+
+func (m *Modifier) SetBehavior(b HeaderSetBehavior) {
```

Comment:
Add comment for exported func
https://github.com/golang/go/wiki/CodeReviewComments#doc-comments